File: tree/BFS.py
class Node:
    def __init__(self, val):
        self.left = None
        self.right = None
        self.key = val

def insert(root, val):
    if root==None:
        root = Node(val)
        return root

    if val < root.key:
        root.left = insert(root.left, val)
    else:
        root.right = insert(root.right, val)
    
    return root

def BFS(root):
    queue = [root]
    queue2 = [root.key]

    while len(queue)>0:
        cur_node = queue.pop(0)
        print("data",cur_node.key)
        # root = queue[0]
        if cur_node.left is not None:
            queue.append(cur_node.left)
            # queue2.append(cur_node.left.key)

        if cur_node.right is not None:
            queue.append(cur_node.right)
            # queue2.append(cur_node.right.key)
        
    # print("queue", queue2)


if __name__=='__main__':
    root = None
    print("Enter n value in BST")
    n = int(input())
    for i in range(n):
        val = int(input())
        root = insert(root, val)
    
    BFS(root)

File: tree/others/BFS.cpp
#include<iostream>
#include<queue>
using namespace std;


struct BSTNode
{
    int data;
    BSTNode *left;
    BSTNode *right;
};

BSTNode* root;


BSTNode* GetNewNode(int data)
{   
    BSTNode *temp = new BSTNode;  
    temp->data = data;
    temp->left = NULL;
    temp->right = NULL;
    return temp;
};


BSTNode* insert(BSTNode* root, int data)
{
    if (root==NULL)
    {
        root = GetNewNode(data);
        return root;
    }

    if (data<=root->data) 
    {
        root->left = insert(root->left, data);
    }
    else
    {
        root->right = insert(root->right, data);
    }

    return root;
}

int BFS(BSTNode* root)
{
    if (root==NULL)
    {
        return 0;
    }

    queue<BSTNode*> q;

    q.push(root);

    while (!q.empty())
    {
        BSTNode* curr = q.front();

        cout<<"data"<<curr->data;
        if (curr->left!=NULL)
        {
            q.push(curr->left);
        }
        
        if (curr->right!=NULL)
        {
            q.push(curr->right);
        }

        q.pop();

    }
    

}

int main()
{
    root=NULL;
    cout<<"Implement Binary Tree"<<endl;
    int n;
    cout<<"Enter n value to be enter in tree ";
    cin>>n;
    for (int i = 0; i < n; i++)
    {
        int data;
        cout<<"Enter data in new node ";
        cin>>data;
        root = insert(root, data);
        cout<<endl;
        cout<<"root data "<<root->data;
        cout<<endl;
    }

    BFS(root);

    return 0;
}